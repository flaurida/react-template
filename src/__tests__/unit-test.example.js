import React from "react";
import { render, waitForElement } from "react-testing-library";

const ExampleComponent = ({ name }) => {
    return <p>{name}.</p>;
};

describe("Example Component", function() {
    test("it renders the name", async () => {
        const name = "Lorem";
        const { container, getByText } = render(<ExampleComponent name={name} />);

        await waitForElement(() => getByText(/Lorem/));

        expect(container).toMatchSnapshot();
    });
});
