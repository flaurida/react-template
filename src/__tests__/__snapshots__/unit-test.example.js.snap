// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example Component it renders the name 1`] = `
<div>
  <p>
    Lorem
    .
  </p>
</div>
`;
